### PR TITLE
Add report functionality for detect-secrets + tests

### PR DESCRIPTION
**Pull request by MVrachev**
This is the reporting functionality for detect_secrets which will
process the results from the detect_secrets detect_secrets scan
and it will create valid GitHub annotation objects for the issues.

All upcoming pull requests to integrate detect_secrets
into Precaution can be found here:
https://github.com/vmware/precaution/issues/209#issuecomment-492964642

Related to: https://github.com/vmware/precaution/issues/209

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

### CODE REVIEW

File: linters/detect_secrets/detect_secrets_annotations.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * @return {Object} returns an annotation object as specified here:
+ * https://developer.github.com/v3/checks/runs/#annotations-object
+ */
+function getAnnotation (issue, filePath) {
+  const path = filePath
+  const startLine = issue.line_number
+  const endLine = startLine
+  const annotationLevel = getAnnotationLevel('HIGH', 'LOW')
+  const title = issue.type
+  const message = `The issue is found by the ${issue.type} plugin in detect-secrets.`
```
Comment (ericwb) [inline linters/detect_secrets/detect_secrets_annotations.js]:
```suggestion
  const message = `The issue was found by the ${issue.type} plugin in detect-secrets.`
```

File: linters/detect_secrets/detect_secrets_report.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Copyright 2019 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+const { getAnnotation } = require('./detect_secrets_annotations')
+const { countIssueLevels } = require('../../annotations_levels')
+
+/**
+ * Process detect-secrets output (generate annotations, count issue levels)
+ * @param {*} results detect-secrets json output
+ */
```
Comment (ericwb) [inline linters/detect_secrets/detect_secrets_report.js]: Add information on return via @return

File: linters/detect_secrets/detect_secrets_annotations.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * @param {*} issue an issue from which the annotation will be build
+ * @param {String} filePath file path where the issue is located
+ * @return {Object} returns an annotation object as specified here:
+ * https://developer.github.com/v3/checks/runs/#annotations-object
+ */
+function getAnnotation (issue, filePath) {
+  const path = filePath
+  const startLine = issue.line_number
+  const endLine = startLine
+  const annotationLevel = getAnnotationLevel('HIGH', 'LOW')
```
Comment (ericwb) [inline linters/detect_secrets/detect_secrets_annotations.js]: Should the severity always be high and confidence always low for these results?

**Comment by MVrachev** [comment]
> So this PR covers only partial functionality. So what other PRs are you planning to submit? It would be good to get in the practice of documenting your planned PRs in the Issue.

I am doing that already: https://github.com/vmware/precaution/issues/160 I was just not sure where to put my planned pull requests.
I mean should I put them in the existing issue tracker as a comment: https://github.com/vmware/precaution/issues/209 or should I create another issue where it will be more visible the progress?

> You PR message added design questions, but the PR is not really a good place for that.

Yes, you are right. I just thought about those design questions just before I submitted the pr and I didn't realize that they are better to be placed in the issue tracker.
I will fix that.

File: linters/detect_secrets/detect_secrets_annotations.js (hunk 3 above)
Comment (MVrachev) [reply linters/detect_secrets/detect_secrets_annotations.js]: That's something we should discuss in our issue tracker
https://github.com/vmware/precaution/issues/209
I just put the severity and confidence Bandit uses when reporting hardcoded credentials.

File: linters/detect_secrets/detect_secrets_report.js (hunk 2 above)
Comment (MVrachev) [reply linters/detect_secrets/detect_secrets_report.js]: I noticed the bandit_report and gosec_report don't have `returns` documentation.
If you approve the way I document the return value of this function I will add the same style for them.

**Comment by MVrachev** [comment]
I moved the design discussion in our issue tracker: https://github.com/vmware/precaution/issues/209#issuecomment-492957066
and I gave planning how many pull requests we need to fully integrate detect-secrets here: https://github.com/vmware/precaution/issues/209#issuecomment-492964642

File: linters/detect_secrets/detect_secrets_annotations.js (hunk 3 above)
Comment (ericwb) [reply linters/detect_secrets/detect_secrets_annotations.js]: You might want to look at https://github.com/lyft/bandit-high-entropy-string. This is a bandit plugin to do detect-secrets like testing.  As part of the plugin, it defines severity and confidence.

**Comment by MVrachev** [comment]
As I explained here https://github.com/vmware/precaution/issues/209#issuecomment-501589208 we decided that we won't use `detect-secret`.

**Pull request closed without merging**